lib/datasource.js

```diff
@@ -632,6 +632,7 @@ DataSource.prototype.setupDataAccess = function(modelClass, settings) {
       if (idProp.generated && forceId !== false) {
         forceId = true;
       }
+      settings.forceId = forceId;
```

Comment:
@rashmihunt I made the changes I had in my mind in 4ccef202d, there were only two tests to fix. Perhaps we misunderstood each other? Anyhow, please take a look at my code and feel free to modify it further as needed.